File: Practice/LinkedList/RemoveLoopFromLinkedList.js
function RemoveLoopFromLinkedList(A) {
  if (A == null || A.next == null) return null;
  let slow_pointer = A;
  let fast_pointer = A;
  while (fast_pointer.next != null && fast_pointer.next.next != null) {
    slow_pointer = slow_pointer.next;
    fast_pointer = fast_pointer.next.next;
    if (slow_pointer == fast_pointer) {
      slow_pointer = A;
      let prev = null;
      while (slow_pointer != fast_pointer) {
        slow_pointer = slow_pointer.next;
        prev = fast_pointer;
        fast_pointer = fast_pointer.next;
      }
      prev.next = null;
      return A;
    }
  }

  return A;
}

File: Revision/LinkedList/RemoveLoopFromLinkedlist.java
public class RemoveLoopFromLinkedlist {
    public ListNode solve(ListNode A) {
        if(A == null) return A;
        ListNode fastPointer = A;
        ListNode slowPointer = A;
        while(fastPointer.next != null && fastPointer.next.next != null){
            slowPointer = slowPointer.next;
            fastPointer = fastPointer.next.next;
            if(slowPointer == fastPointer){
                ListNode prev = null;
                slowPointer = A;
                while(slowPointer != fastPointer){
                    prev = fastPointer;
                    slowPointer = slowPointer.next;
                    fastPointer = fastPointer.next;
                }
                prev.next = null;
                return A;
            }
        }

        return null;
    }
}
